modules/ITK/cli/utils/statismo-build-gp-model-kernels.h

```diff
@@ -0,0 +1,131 @@
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED
+ * TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+ * LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+ * NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ */
+
+#include <map>
+#include <string>
+
+#include <boost/algorithm/string.hpp>
+#include <boost/math/distributions/normal.hpp>
+
+#include <itkImage.h>
+#include <itkMesh.h>
+
+#include "KernelCombinators.h"
+#include "Kernels.h"
+
+/*
+You can add your kernels here:
+	1. Either #include your kernel or put the class in here
+	2. Write a function in this file that creates your kernel (create it on the heap and NOT on the stack; it will be deleted, don't worry)
+	   Your function will receive a vector containing the kernel arguments (vector<string>). You have to parse them or thrown an itk::ExceptionObject if something goes wrong.
+	3. Add your kernel to the kernel map (do this in the function createKernelMap(); your kernel name has to be lowercase)
+	4. Document your kernel in the statismo-build-gp-model.md file. (Examples for kernels with multiple parameters are in there, but currently commented out)
+*/
+
+
+const unsigned Dimensionality3D = 3;
+typedef itk::Mesh<float, Dimensionality3D> DataTypeShape;
+typedef itk::Image< itk::Vector<float, Dimensionality3D>, Dimensionality3D > DataType3DDeformation;
+const unsigned Dimensionality2D = 2;
+typedef itk::Image< itk::Vector<float, Dimensionality2D>, Dimensionality2D > DataType2DDeformation;
+
+struct KernelContainer {
+    const statismo::ScalarValuedKernel<DataTypeShape::PointType>* (*createKernelShape)(std::vector<std::string> kernelArgs);
+    const statismo::ScalarValuedKernel<DataType3DDeformation::PointType>* (*createKernel3DDeformation)(std::vector<std::string> kernelArgs);
+    const statismo::ScalarValuedKernel<DataType2DDeformation::PointType>* (*createKernel2DDeformation)(std::vector<std::string> kernelArgs);
+};
+typedef std::map<std::string, KernelContainer> KernelMapType;
+KernelMapType kernelMap;
+
+
+template <class TPoint>
+class GaussianKernel : public statismo::ScalarValuedKernel<TPoint> {
+  public:
+    typedef typename  TPoint::CoordRepType CoordRepType;
+    typedef vnl_vector<CoordRepType> VectorType;
+
+
+    GaussianKernel(double sigma) : m_sigma(sigma), m_sigma2(1.0 / (sigma * sigma)) {
+
+    }
+
+    inline double operator()(const TPoint& x, const TPoint& y) const {
+        VectorType xv = x.GetVnlVector();
+        VectorType yv = y.GetVnlVector();
+
+        VectorType r = yv - xv;
+
+        return exp(-dot_product(r, r) * m_sigma2);
```

Comment:
This line could be replaced with the following code 

> return boost::math::pdf(boost::math::normal(0, sigma), sqrt(dot_product(r, r))); 

The function above returns different results though: *exp(-dot_product(r, r) * m_sigma2);* doesn't apply the factor of 1/2 to *m_sigma2* and doesn't scale the function so that the integral over the entire thing is = 1.

Should I replace it despite the different values it will return or leave it?

[boost::math::normal(...) would of course be created only once instead of every call]